### PR TITLE
feat:  buildkite bare minimun cli reporter

### PR DESCRIPTION
**Pull request by shahzad31**
bare minimun cli reporter

### CODE REVIEW

**Review by vigneshshanmugam — COMMENTED** [review]
Looks good mostly, just added few comments for errors and output formats.

File: src/reporters/build_kite_cli.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import { indent, now, symbols } from '../helpers';
+import {
+  JourneyEndResult,
+  JourneyStartResult,
+  StepEndResult,
+} from '../common_types';
+import { Journey, Step } from '../dsl';
+import { renderError } from './reporter-util';
+
+export default class BuildKiteCLIReporter extends BaseReporter {
```
Comment (vigneshshanmugam) [inline src/reporters/build_kite_cli.ts:39]: This is how I am seeing it, Are you okay with this format? If you have the output from build kite once these suggested changes are pushed, please do share it. 

Screenshot 2023-12-06 at 3 22 03 PM

File: src/reporters/build_kite_cli.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  override async onJourneyEnd(
+    journey: Journey,
+    { error, start, end, status }: JourneyEndResult
+  ) {
+    const { failed, succeeded, skipped } = this.metrics;
+    const total = failed + succeeded + skipped;
+    if (total === 0 && error) {
+      this.write(renderError(error));
+    }
+    const message = `${symbols[status]} Took  (${renderDuration(
```
[the remark this answers is not in the document]
Comment (shahzad31) [reply src/reporters/build_kite_cli.ts:75]: Yes, it makes sense to me

File: src/reporters/build_kite_cli.ts (hunk 1 above)
Comment (shahzad31) [reply src/reporters/build_kite_cli.ts:39]: yes, that looks good to me

File: src/index.ts [hunk 3]
```diff
@@ -63,7 +63,16 @@ export type {
 export type { default as Runner } from './core/runner';
 export type { Reporter, ReporterOptions } from './reporters';
 
-export type { SyntheticsConfig } from './common_types';
+export type {
+  SyntheticsConfig,
```
Comment (vigneshshanmugam) [inline src/index.ts:67]: nit: expose StartEvent

File: src/reporters/build_kite_cli.ts [hunk 4]
```diff
@@ -70,42 +62,44 @@ export default class BuildKiteCLIReporter extends BaseReporter {
     const { failed, succeeded, skipped } = this.metrics;
     const total = failed + succeeded + skipped;
     if (total === 0 && error) {
-      this.write(renderError(error));
+      this.write(renderError(serializeError(error)));
```
Comment (vigneshshanmugam) [inline src/reporters/build_kite_cli.ts:65]: there is a indentation issue here. I would just do the same here by calling `super.onJourneyEnd` and then add the `Took duration message`

**Review by vigneshshanmugam — APPROVED** [review]
LGTM 🎉

**Pull request merged**